apps/abstraxion-dashboard/components/AbstraxionGrant/index.tsx

```diff
@@ -50,6 +60,67 @@ export const AbstraxionGrant = ({
     }
   });
 
+  const generateStakeGrant = (
+    granter: string,
+  ): Array<{
+    typeUrl: string;
+    value: MsgGrant;
+  }> => {
+    const timestampThreeMonthsFromNow = Math.floor(
+      new Date(new Date().setMonth(new Date().getMonth() + 3)).getTime() / 1000,
+    );
+
+    // Need to grant MsgWithdrawDelegatorReward
+    const msgGrant = {
+      typeUrl: MsgGrant.typeUrl,
+      value: MsgGrant.fromPartial({
+        grant: {
+          authorization: {
+            typeUrl: GenericAuthorization.typeUrl,
+            value: GenericAuthorization.encode(
+              GenericAuthorization.fromPartial({
+                msg: MsgWithdrawDelegatorReward.typeUrl,
+              }),
+            ).finish(),
+          },
+          expiration: {
+            seconds: BigInt(timestampThreeMonthsFromNow),
+          },
+        },
+        grantee,
+        granter,
+      }),
+    };
+
+    return [
+      AuthorizationType.AUTHORIZATION_TYPE_DELEGATE,
+      AuthorizationType.AUTHORIZATION_TYPE_UNDELEGATE,
+      AuthorizationType.AUTHORIZATION_TYPE_REDELEGATE,
+    ]
+      .map((authorizationType) => ({
+        typeUrl: MsgGrant.typeUrl,
+        value: MsgGrant.fromPartial({
+          grant: {
+            authorization: {
+              typeUrl: StakeAuthorization.typeUrl,
+              value: StakeAuthorization.encode(
+                StakeAuthorization.fromPartial({
+                  authorizationType:
+                    AuthorizationType.AUTHORIZATION_TYPE_DELEGATE,
+                }),
+              ).finish(),
+            },
+            expiration: {
+              seconds: BigInt(timestampThreeMonthsFromNow),
+            },
+          },
+          grantee,
+          granter,
+        }),
+      }))
+      .concat(msgGrant);
```

Comment:
I wonder if it would also be possible to add the `/cosmos.feegrant.v1beta1.BasicAllowance` allowance, using `/cosmos.feegrant.v1beta1.MsgGrantAllowance`, as an extra message, to allow the application key to do staking even if it doesn't own tokens. Or is that flow intended with a different approach (thinking maybe using contracts)?